libobs/obs-encoder.h

```diff
@@ -280,6 +293,12 @@ struct obs_encoder_info {
 			       uint64_t lock_key, uint64_t *next_key,
 			       struct encoder_packet *packet,
 			       bool *received_packet);
+
+	bool (*encode_texture2)(void *data, struct encoder_texture *texture,
+				int64_t pts, uint64_t lock_key,
```

Comment:
If we're introducing an encoder_texture struct, it seems like `pts`, `lock_key`, and `next_key` should be values of that struct.

Reply:
pts sure, the next_key is technically an output variable so making that part of the struct feeels kind of ugly.

Reply:
im not sure i agree with any of them, its just that we dont share/reuse textures that all of these happen to have the same lifetimes.

Reply:
I see. What I have been wondering if perhaps we can move at least the pts to a struct similar to `encoder_frame`, the motivation here is that for the multi-encoder streaming Twitch is doing and #10096 we'll need a way to force IDR frames. 
For non-texture encoders we can simply extend the existing struct without breaking ABI, but for GPU encoders that isn't possible. So perhaps we can make this API more future-looking by adding a means to add more information later without requiring a new ABI?

I realise this is getting into bikeshedding territory, I won't block this PR on that, just wanted to hear your opinion.